app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1543,4 +1381,25 @@ public boolean isBackEnabled() {
         return false;
     }
 
+    /**
+     * For Testing purposes only
+     */
+    public SessionNavigator getSessionNavigator() {
+        if (BuildConfig.DEBUG) {
+            return sessionNavigator;
+        } else {
+            throw new RuntimeException("On principal of design, only meant for testing purposes");
+        }
+    }
+
+    /**
+     * For Testing purposes only
+     */
+    public void setFormAndDataSyncer(FormAndDataSyncer formAndDataSyncer) {
+        if (BuildConfig.DEBUG) {
+            this.formAndDataSyncer = formAndDataSyncer;
+        } else {
+            throw new RuntimeException("On principal of design, only meant for testing purposes");
+        }
```

Comment:
Sad that cruft sometimes comes along with tests. Definitely interested in hearing opinions on better ways to do this.